File: GreatestCommonDivisor/C++/GreatestCommonDivisor.cpp
#include <iostream>

// Custom gcd function
int gcd(int a, int b) {
    while (b != 0) {
        int temp = b;
        b = a % b;
        a = temp;
    }
    return a;
}

int main() {
    std::cout << "The greatest common divisor of 12 and 18 is " << gcd(12, 18) << " !\n";
    return 0;
}

File: GreatestCommonDivisor/Java/GreatestCommonDivisor.java
public class GreatestCommonDivisor {

    public static long GreatestCommonDivisor(long a, long b) {
        long factor = Math.min(a, b);
        for (long loop = factor; loop > 1; loop--) {
            if (a % loop == 0 && b % loop == 0) {
                return loop;
            }
        }
        return 1;
    }

    public static void main(String[] args) {
        long a = 12;
        long b = 18;
        System.out.println("The greatest common divisor of " + a + " and " + b + " is " + GreatestCommonDivisor(a, b));
    }
}